pkg/api/pod/util_test.go

```diff
@@ -176,6 +176,10 @@ func collectResourcePaths(t *testing.T, resourcename string, path *field.Path, n
 	case reflect.Ptr:
 		resourcePaths.Insert(collectResourcePaths(t, resourcename, path, name, tp.Elem()).List()...)
 	case reflect.Struct:
+		// Specifically skip ObjectMeta because it has recursive types
```

Comment:
This is not a good justification for this skip. If the problem were recursive types, the solution would be cycle detection.

Fortunately, there is a better justification: ObjectMeta is generic and therefore should never have a field with a specific resource's name. (Assuming I understand what this function is doing.)